articles/153_roslaunch_remote_launch.md

```diff
@@ -0,0 +1,94 @@
+This component won't be necessary for basic execution of remote nodes, but would allow remote system to persist beyond the life of the initial LaunchService.
+The topics and services will allow remote tools to view and manage the processes running remotely.
+The LaunchServiceNode allows users to monitor and manage the status and lifecycles of nodes across systems, and enables the system to continue running if the machine used to start a system becomes disconnected.
+
+
+
+## New Classes
+
+#### launch.descriptions.Machine
+##### Members
+Members are defined by child classes, since different protocols require different types of information
+
+#### launch.descriptions.SSHMachine
+Contains information needed to run a process on another machine via ssh
+##### Members
+| Name | Type | Description |
+|---|---|---|
+| hostname   | String | Hostname of the machine. |
+| port       | int | Port on the host to connect to (default 22) |
+| ssh\_keys  | String | Path to ssh key file |
+| passphrase | String | Optional passphrase if the key uses one. |
+
+#### launch.actions.ExecuteRemote
+Base class for actions that use connection information contained in `machine` to start a process on a remote machine.
+##### Members
+| Name | Type | Description |
+|---|---|---|
+| process\_description  |`launch.descriptions.Executable` | An object describing the process to be executed.|
+| shell                 | boolean | If True, a shell is used to execute the process.|
+| sigterm\_timeout     | List[Substitution] | Time until shutdown should escalate to `SIGTERM`, as a string or a list of strings and `Substitution`s to be resolved at runtime, defaults to the `LaunchConfiguration` called `sigterm_timeout` |
+| sigkill\_timeout     | List[Substitution] | Time until escalating to `SIGKILL` after `SIGTERM`, as a string or a list of strings and `Substitution`s to be resolved at runtime, defaults to the `LaunchConfiguration` called `sigkill_timeout` |
+| emulate\_tty         | boolean | Emulate a tty (terminal), defaults to `False`, but can be overridden with the `LaunchConfiguration` called `emulate_tty`, the value of which is evaluated as true or false according to `evaluate_condition_expression`. |
+| prefix               | List[Substitution] | A set of commands/arguments to preceed the `cmd`, used for things like `gdb`/`valgrind` and defaults to the `LaunchConfiguration` called `launch-prefix` |
+| output               | ?? | Configuration for process output logging. Defaults to `log` i.e. log both `stdout` and `stderr` to launch main log file and stderr to the screen. |
+| output\_format       | ?? | For logging each output line, supporting `str.format()` substitutions with the following keys in scope: `line` to reference the raw output line and `this` to reference this action instance. |
+| log\_cmd             | boolean | If `True`, prints the final cmd before executing the process, which is useful for debugging when substitutions are involved. |
```

Comment:
maybe should have a note that this will avoid printing any key passphrases or other sensitive information

Reply:
Oh, good point, I will include a note about that.